Core/Tornado/Server/TornadoService.py

```diff
@@ -0,0 +1,538 @@
+      # before authentication we return 401 UNAUTHORIZED instead of 403 FORBIDDEN
+      self.reportUnauthorizedAccess(HTTPErrorCodes.HTTP_UNAUTHORIZED)
+
+    try:
+      hardcodedAuth = getattr(self, 'auth_' + self.method)
+    except AttributeError:
+      hardcodedAuth = None
+
+    self.authorized = self._authManager.authQuery(self.method, self.credDict, hardcodedAuth)
+    if not self.authorized:
+      self.reportUnauthorizedAccess()
+
+  @gen.coroutine
+  def post(self):  # pylint: disable=arguments-differ
+    """
+    HTTP POST, used for RPC
+      Call the remote method, client may send his method via "method" argument
+      and list of arguments in JSON in "args" argument
+    """
+
+    # Execute the method
+    # First argument is "None", it's because we let Tornado manage the executor
+    retVal = yield IOLoop.current().run_in_executor(None, self.__executeMethod)
+
+    # Tornado recommend to write in main thread
+    # TODO CHRIS READ THAT https://www.tornadoweb.org/en/branch5.1/web.html#thread-safety-notes
+    self.__write_return(retVal.result())
+    self.finish()
+
+  # This nice idea of streaming to the client cannot work because we are ran in an executor
+  # and we should not write back to the client in a different thread.
+  # See https://www.tornadoweb.org/en/branch5.1/web.html#thread-safety-notes
+  # def export_streamToClient(self, filename):
+  #   # https://bhch.github.io/posts/2017/12/serving-large-files-with-tornado-safely-without-blocking/
+  #   #import ipdb; ipdb.set_trace()
+  #   # chunk size to read
+  #   chunk_size = 1024 * 1024 * 1  # 1 MiB
+
+  #   with open(filename, 'rb') as f:
+  #     while True:
+  #       chunk = f.read(chunk_size)
+  #       if not chunk:
+  #         break
+  #       try:
+  #         self.write(chunk)  # write the chunk to response
+  #         self.flush()  # send the chunk to client
+  #       except StreamClosedError:
+  #         # this means the client has closed the connection
+  #         # so break the loop
+  #         break
+  #       finally:
+  #         # deleting the chunk is very important because
+  #         # if many clients are downloading files at the
+  #         # same time, the chunks in memory will keep
+  #         # increasing and will eat up the RAM
+  #         del chunk
+  #         # pause the coroutine so other handlers can run
+  #         yield gen.sleep(0.000000001)  # 1 nanosecond
+
+  #   return S_OK()
+
+  @gen.coroutine
+  def __executeMethod(self):
+    """
+      Execute the method called, this method is executed in an executor
+      We have several try except to catch the different problem who can occurs
+
+      - First, the method does not exist => Attribute error, return an error to client
+      - second, anything happend during execution => General Exception, send error to client
+    """
+
+    # getting method
+    try:
+      method = getattr(self, 'export_%s' % self.method)
```

Comment:
Just add a comment that this is here for compatibility purpose.

Reply:
sorry I am not completely sure I understand what you mean. Compatibility with what ? The fact that we kept the `export_` prefix ?

Reply:
Putting `export_` for signaling a RPC call (which will become a route, IIUC in HTTP) is just the "DISET" way. We keep for backward compatibility, nothing else.

Reply:
I'll add the comment yes. 
No, it's not a route. There is one route per service. The rest, including the method name, is part of the post attributes.
Making one route per call would be nice, and go towards REST, but since REST would anyway require many more api change, we did not bother

Reply:
Ah, the fact that it's only the service and not the function that is becoming a route was not clear to me. This should be pointed out in the docs and in the pres for the next BiLD.

Reply:
OK, I will better document al that